diff --git a/node_modules/react-native-reanimated/plugin.js b/node_modules/react-native-reanimated/plugin.js
index 1bf9f1d..25e1c51 100644
--- a/node_modules/react-native-reanimated/plugin.js
+++ b/node_modules/react-native-reanimated/plugin.js
@@ -1,10 +1,7 @@
 'use strict';
 
-const generate = require('@babel/generator').default;
+const { generate, traverse, parse, visitors, getPlugin } = require('@expo/babel-standalone');
 const hash = require('string-hash-64');
-const { visitors } = require('@babel/traverse');
-const traverse = require('@babel/traverse').default;
-const parse = require('@babel/parser').parse;
 
 /**
  * holds a map of function names as keys and array of argument indexes as values which should be automatically workletized(they have to be functions)(starting from 0)
@@ -503,7 +500,7 @@ const PLUGIN_BLACKLIST_NAMES = ['@babel/plugin-transform-object-assign'];
 
 const PLUGIN_BLACKLIST = PLUGIN_BLACKLIST_NAMES.map((pluginName) => {
   try {
-    const blacklistedPluginObject = require(pluginName);
+    const blacklistedPluginObject = getPlugin(pluginName);
     // All Babel polyfills use the declare method that's why we can create them like that.
     // https://github.com/babel/babel/blob/32279147e6a69411035dd6c43dc819d668c74466/packages/babel-helper-plugin-utils/src/index.js#L1
     const blacklistedPlugin = blacklistedPluginObject.default({
